sycl/doc/extensions/supported/sycl_ext_intel_queue_immediate_command_list.asciidoc

```diff
@@ -0,0 +1,160 @@
+is chosen by the implementation.
+
+When commands are submitted to a SYCL queue that uses the Level Zero backend,
+those commands can be submitted to the hardware in one of two ways:
+either through an immediate command list or through a standard command queue.
+Commands submitted through an immediate command list are immediately submitted
+to the device while commands submitted through a standard command queue may be
+batched with other commands before they are submitted. By default the
+implementation chooses a method that works best for most workloads.
+
+In most cases, applications should rely on the default behavior.
+However, sometimes it is advantageous for the application to choose one method
+or the other. This extension provides a way for applications to select either
+of these two methods via a queue property.
+
+For example, when kernel runtimes are very short, the submission time on the
+host may be as long or longer than the actual runtime of the kernel. In this
+case, doing batched submissions may be preferable so that the submission
+overhead is amortized over a number of kernel executions.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_QUEUE_IMMEDIATE_COMMAND_LIST` to one of the values
+defined in the table below.  Applications can test for the existence of this
+macro to determine if the implementation supports this feature, or
+applications can test the macro's value to determine which of the
+extension's features the implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+=== New queue properties
+This extension adds the following new properties that can be used when
+constructing a queue object.
+
+```c++
+namespace ext {
+namespace intel {
+
+// Use immediate command lists
+property_list P1{property::queue::immediate_command_list()};
+queue Q1{P1};       
+
+// Do not use immediate command lists
+property_list P2{property::queue::no_immediate_command_list()};
+queue Q2{P2};
+
+...
+
+} // namespace intel
+} // namespace ext
+```
+
+
+Both properties are hints, which are ignored unless the backend is Level Zero.
+
+The property `immediate_command_list` requests that the implementation use an
+immediate command list when commands are submitted to this queue. As a result,
+these commands are submitted immediately to the device.
+
+The property `no_immediate_command_list` requests that the implementation use
+a standard command queue instead of an immediate command list. As a result,
+commands submitted to this queue may be batched with other commands before
+being submitted to the device.
+
+These two properties are mutually exclusive. Constructing a queue with both
+properties causes the constructor to throw a synchronous exception with
+the `errc::invalid` error code.
+
```

Comment:
It is useful to add an example usage, though.  If you want to do this, add a new top-level section named "Example":

> == Example
>
> ```
> #include <sycl/sycl.hpp>
>
> namespace syclintel = sycl::ext::intel;
>
> int main() {
>   // Use immediate command lists
>   sycl::queue q1{syclintel::property::queue::immediate_command_list{}};
>
>   // Do not use immediate command lists
>   sycl::queue q1{syclintel::property::queue::no_immediate_command_list{}};
> }

Note the following:

* This example is not defined inside the `ext::intel` namespace as you had above.  User's don't write code in that namespace.
* We generally create a namespace alias in these examples as I show above for `syclintel`.
* There's no need to create a separate `property_list` object as you had in your example.  The syntax I show above is a common shorthand.  What you had is legal, though, so if you want to show the more verbose version, that's OK too.
* Be sure to show the full namespace for APIs as I have in the example above.  We want these examples to be super clear about which namespace to use for each API.
* When possible, it's better to show examples that are complete programs as I have above.  This allows you to cut-and-paste the example into an editor and compile the code.